### PR TITLE
Submission miscellaneous fixes

### PR DESCRIPTION
**Pull request by atarix83**
This PR solves some small miscellaneous issues for submission module.
Here the the most important changes:

- during submission show only authorized collections in the collection dropdown list 
- Add hint message to form fields, hint message it's already present in rest configuration but it's displayed on the form
- fix issues [DS-4280](https://jira.duraspace.org/browse/DS-4280) and [DS-4279](https://jira.duraspace.org/browse/DS-4279) 
- during workspaceitem edit hide upload section on init when section is not mandatory and there are no file uploaded in the submission
- other small bug fixes

### CODE REVIEW

File: src/app/core/data/collection-data.service.ts [hunk 1]
```diff
@@ -40,6 +41,22 @@ export class CollectionDataService extends ComColDataService<Collection> {
     super();
   }
 
+  /**
+   * Get all collections whom user has authorization to submit to by community
```
[the remark this answers is not in the document]
Comment (atarix83) [reply src/app/core/data/collection-data.service.ts]: Fixed

**Comment by atarix83** [comment]
@artlowel  thanks for the review

> I can still reproduce https://jira.duraspace.org/browse/DS-4279 however, although the edit button does work now.

This is how I see the buttons, there is space between all buttons, is not the same for you?

Schermata del 2019-07-26 09-54-13

> Are you using very large page sizes when fetching collections (here and here) in order to bypass this issue: #373?

No this is not a permanent solution, is only to avoid to miss some collection in the list. I think that a better solution until we have a suitable rest endpoint, should be to remove community of the collection from the list, this solution would make the loading less heavy

> The warning in the upload files panel will still mention the original collection. It wouldn't surprise me if this was a pre-existing issue though.
I've fixed the issue

**Comment by artlowel** [comment]
@atarix83 I see the same buttons, but https://jira.duraspace.org/browse/DS-4279 says that in the first step you should only be able to see accept and reject buttons (so no edit), and in the last step only edit and accept buttons (so no reject).

>I think that a better solution until we have a suitable rest endpoint, should be to remove community of the collection from the list, this solution would make the loading less heavy

Yeah that sounds like a good idea 👍

**Comment by atarix83** [comment]
@artlowel 
> I think that a better solution until we have a suitable rest endpoint, should be to remove community of the collection from the list, this solution would make the loading less heavy

this has a side effect, in case of more collection belonging to different communities with same name, the user would not know to which collection it's submitting to

> says that in the first step you should only be able to see accept and reject buttons (so no edit), and in the last step only edit and accept buttons (so no reject).

Based on how workspaceitems/workflowitems endpoints are currently implemented, I think it is not possible to have this distinction, so I'd leave out resolution of [https://jira.duraspace.org/browse/DS-4279](https://jira.duraspace.org/browse/DS-4279) from this PR

**Comment by artlowel** [comment]
Retested, and everything looks good. Thanks @atarix83 👍

**Comment by artlowel** [comment]
@tdonohue I've noticed that once, when I first tried this PR, but wasn't able to reproduce it since.

**Comment by tdonohue** [comment]
This PR needs rebasing, as it now has merge conflicts.

**Comment by atarix83** [comment]
@tdonohue 
I've investigated on the issue reported.

It seems that is due to the way the collection list is retrieved when submission page is loading.
In order to be able to view only the collection the user can submit to I've changed the [submission-form-collection.component.ts](https://github.com/4Science/dspace-angular/blob/submission-miscellaneous-fixes/src/app/submission/form/collection/submission-form-collection.component.ts).
Indeed even if the submission page seems to be loaded, the `submission-form-collection.component.ts` is still retrieving the collection list by using [getAuthorizedCollectionByCommunity](https://github.com/4Science/dspace-angular/blob/submission-miscellaneous-fixes/src/app/core/data/collection-data.service.ts#L66). Unfortunately to have the entire collection list we need the entire community list, but this cause an extremely expensive REST call (because of fetching all logos and licenses) that is the cause the delay reported.

To fix this behaviour there are these possible solution:
- Revert all the changes and show always all collection list, but this should cause error when submitting in an unauthorized collection.
- Fetch collection list with [getAuthorizedCollection](https://github.com/4Science/dspace-angular/blob/67b09e379427810e44472591a6c15e5dc1846aaa/src/app/core/data/collection-data.service.ts#L51), in this case we can't see the parent community and we can't discern between collections with the same name but in different community
- as suggested by @abollini we should need to add, on REST side, a `parent` link to the collection model, in order to retrieve parent community information from collection   

@tdonohue @tdonohue 
Please let me know what do you think about it

**Comment by tdonohue** [comment]
@atarix83 : I understand the problem now. This sounds like a performance issue that could be helped by REST API projections, as well as the suggestion from @abollini to perhaps add a "parent" link (which might be useful for other purposes too).  

👍 I'm OK with merging this as-is (with the known performance issues).  However, before we do so, I'd like someone to log a bug ticket describing the problem and possible REST API solution(s).  That way we can get this scheduled/assigned, and possibly link it up to similar work (like Projections).  (Unfortunately, I'm adding this note just before heading out of the office for next week, so I won't be able to create this JIRA ticket myself.  But, if someone else can do so, I'm fine with moving this PR forward as-is.)

**Pull request merged**